src/main/java/com/wedevol/xmpp/server/CcsClient.java

```diff
@@ -487,7 +504,7 @@ public void connectionClosedOnError(Exception e) {
     @Override
     public void connectionClosed() {
         logger.info("Connection closed. The current connectionDraining flag is: {}", isConnectionDraining);
-        if (isConnectionDraining) {
+        if (isConnectionDraining || isPresenceUnavailable) {
```

Comment:
do we need a back-off on reconnect?

Reply:
It only backs-off every time it cannot connect. So in this case it should connect immediately but will back-off if it cannot.